src/js/helpers/timeline/declarationHelper-groups.spec.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2020 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { extractGroupData, reduceSubgroups } from './declarationHelper';
+
+test('Group with parent information is correctly transform into group with subgroups information', () => {
+  const alreadyReducedData = { parent: [] };
+  const currentData = { group: 'child', subgroupOf: 'parent' };
+  const expectedData = { parent: ['child'], child: [] };
```

Comment:
You have `'child'` and and `child` in the same line - can be misleading. The same with parent with `parent`.

How about something like that:
```
  const parentGroupName = 'parentGroup';
  const childGroupName = 'childGroup'
  const alreadyReducedData = { parent: [] };
  const currentData = { group: childGroupName, subgroupOf: parentGroupName };
  const expectedData = { parent: [childGroupName], child: [] };
```

Reply:
It cannot be the way you suggested, but right - it is misleading. `'child'` + `child` and `'parent'` + `parent` have the same name because they represent the same groups and are arbitrary.  I'll change those names to something like this to make it clearer:

```
const alreadyReducedData = { parentName: [] };
const currentData = { group: 'childName', subgroupOf: 'parentName' };
const expectedData = { parentName: ['childName'], childName: [] };
```